blockchain/stake/staketx_test.go

```diff
@@ -635,14 +647,235 @@ func TestSSGenErrors(t *testing.T) {
 	ssgentestGenOutputUntagged.SetTree(wire.TxTreeStake)
 	ssgentestGenOutputUntagged.SetIndex(0)
 
-	err = stake.CheckSSGen(ssgentestGenOutputUntagged.MsgTx())
+	err = stake.CheckSSGen(ssgentestGenOutputUntagged.MsgTx(), noTreasury)
 	if !errors.As(err, &serr) || serr.GetCode() != stake.ErrSSGenBadGenOuts {
 		t.Errorf("CheckSSGen should have returned %v but instead returned %v",
 			stake.ErrSSGenBadGenOuts, err)
 	}
-	if stake.IsSSGen(ssgentestGenOutputUntagged.MsgTx()) {
+	if stake.IsSSGen(ssgentestGenOutputUntagged.MsgTx(), noTreasury) {
+		t.Errorf("IsSSGen claimed an invalid ssgen is valid")
+	}
+
+	// ---------------------------------------------------------------------------
+	// Tresury enabled
+
```

Comment:
Could use a test for a positive path first I think. Or is `TestSSGenTreasuryVotes` good enough?

Also could use a test for an invalid version `ErrSSGenInvalidTxVersion`.

Also as I stated earlier, no error or test for a push value that doesn't reflect the actual amount of data. Maybe that's assumed to be caught during script parsing?

Reply:
It is indeed caught but I did add a test and an extra guard to prove that point.